### PR TITLE
Migrate the recording_json table to use JSONB

### PR DESCRIPTION
**Pull request by mayhem**
# Summary

* This is a…
    * [x] Bug fix
    * [ ] Feature addition
    * [x] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

The recording_json table was in the older JSON format and had invalid JSON records. This update script re-writes the table in a single transaction, re-creating all of the ids, FKs and indexes as needed.

We should be able to run this script at any time without interrupting other services.

However, this is based on the assumption that JSON and JSONB columns act in exactly the same way and no other existing code needs to be updated. Is this assumption valid?

### CODE REVIEW

**Comment by mayhem** [comment]
This is the table after being fixed:

https://gist.github.com/mayhem/5856572e5865bb84473cca95b33762a7

**Comment by paramsingh** [comment]
>no other existing code needs to be updated

We don't really have any code here that runs in production other than this (https://github.com/metabrainz/messybrainz-server/blob/master/messybrainz/__init__.py#L7), so should be good to go imo. I don't think we have any JSON specific usages in the codebase re code in production, not sure about kartikeya's work. 

Also, should we base this off the `production` branch?

**Comment by alastair** [comment]
> However, this is based on the assumption that JSON and JSONB columns act in exactly the same way and no other existing code needs to be updated. Is this assumption valid?

The failure that you saw when selecting features would occur as an exception on insertion, so we have to be sure we catch that properly. Looking at the method that @paramsingh posted, I have no idea why we try it 3 times without altering the data

File: messybrainz/db/testing.py [hunk 1]
```diff
@@ -64,6 +65,10 @@ def drop_tables(self):
         db.run_sql_script(os.path.join(ADMIN_SQL_DIR, 'drop_tables.sql'))
 
 
+    def drop_db(self):
+        db.run_sql_script_without_transaction(os.path.join(ADMIN_SQL_DIR, 'drop_db.sql'))
```
Comment (alastair) [inline messybrainz/db/testing.py:69]: (I thought I wrote this comment, but I don't see it any more)
In AB and LB we create the database once before we run the tests, and then the test setup deletes and recreates the tables. This speeds up the runtime of the tests a bit.
 - a thought - postgres will refuse to drop a database if there are active connections to it. Does sqlalchemy keep any active connections around that might cause a problem with this?

**Comment by mayhem** [comment]
I tried to change the review base against production, but it blew up with 172 files. It might just be best to review and approve the PR and then merge it and also merge it into production for release.

**Comment by paramsingh** [comment]
>I tried to change the review base against production, but it blew up with 172 files. It might just be best to review and approve the PR and then merge it and also merge it into production for release.

Sounds good to me.

File: messybrainz/db/testing.py (hunk 1 above)
Comment (paramsingh) [reply messybrainz/db/testing.py:69]: This seems like out of scope for this PR, i just did the simplest thing to fix it for now, I've opened a ticket and will fix this in a different PR: https://tickets.metabrainz.org/browse/LB-455

**Comment by mayhem** [comment]
OK, where do we stand on this PR then -- anything else to address?

**Review by paramsingh — APPROVED** [review]
Looks good to me, we should merge to production after merging this and move forward.

**Comment by mayhem** [comment]
Yep -- can you please do the honors of preparing the updated production branch and then running the upgrade script in production? Perhaps test it one last time on another server for sanity checking? I could easily produce a VM for this purpose...

**Pull request merged**